backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs

```diff
@@ -234,6 +234,50 @@ module MatchPattern =
     | PT.MPVariable(_, name) ->
       RT.MPVariable rc, (symbols |> Map.add name rc), rc + 1
 
+    | PT.MPOr(_, patterns) ->
+      // transform each pattern into a tuple containing:
+      // - the converted pattern (to RT)
+      // - the symbols used in the pattern
+      // - the register count needed by the pattern
+      let patternsWithSymbols =
+        patterns
+        |> List.map (fun pat ->
+          let pat, patSymbols, patRc = toRT Map.empty rc pat
+          (pat, patSymbols, patRc))
+
+      // find the highest register count needed across all patterns
+      let maxRc = patternsWithSymbols |> List.map (fun (_, _, rc) -> rc) |> List.max
```

Comment:
this as-is is fine, but FYI there's also `List.maxBy` if you wanted this to be a tiny bit shorter

Reply:
I think it turned out a little longer
```
let maxRc =
        patternsWithSymbols
        |> List.maxBy (fun (_, _, rc) -> rc) // it returns a (MatchPattern * Map<string, Register> * int)
        |> (fun (_, _, rc) -> rc)
```
I never really used it, thanks for mentioning it. Should we add it to the list of Stdlib functions that we want to add to Darklang?